src/invidious/helpers/webvtt.cr

```diff
@@ -0,0 +1,67 @@
+# Namespace for logic relating to generating WebVTT files
+#
+# Probably not compliant to WebVTT's specs but it is enough for Invidious.
+module WebVTT
+  # A WebVTT builder generates WebVTT files
+  private class Builder
+    def initialize(@io : IO)
+    end
+
+    # Writes an vtt cue with the specified time stamp and contents
+    def cue(start_time : Time::Span, end_time : Time::Span, text : String)
+      timestamp(start_time, end_time)
+      @io << text
+      @io << "\n\n"
+    end
+
+    private def timestamp(start_time : Time::Span, end_time : Time::Span)
+      add_timestamp_component(start_time)
+      @io << " --> "
+      add_timestamp_component(end_time)
+
+      @io << '\n'
+    end
```

Comment:
Why making a dedicated function for that? Can't it be integrated into `cue()`?

Reply:
No reason in particular. Imo it feels cleaner to treat the timestamp as a single entity in `#cue` rather than having it be made out of two timestamp components and an arrow; thus a function. Do you want me to get rid of `#timestamp`?

Reply:
Oh okay. Keep it like that, then!